src/AMOEBA/amoeba_convolution.cpp

```diff
@@ -326,15 +326,23 @@ FFT_SCALAR *AmoebaConvolution::pre_convolution_3d()
     cfft[n++] = ZEROF;
   }
 
+  double time0,time1;
```

Comment:
Do we really need such detailed internal profiling throughout the Amoeba code for production use?
If no, it could be "hidden" in `#if DEBUG_AMOEBA` sections or fully removed.
If yes, you should be using `platform::walltime()` instead of `MPI_Wtime()` and there should be no unconditional barriers. At the very least they should be prefixed with `if timer->has_sync()` to only synchronize processes when the user requests it to timings.

Reply:
Thanks for the suggestion. Will work on this.